packages/template-retail-react-app/app/hooks/use-product-view-modal.js

```diff
@@ -73,18 +70,6 @@ export const useProductViewModal = (initialProduct) => {
         }
     }, [])
 
-    useEffect(() => {
-        if (variant) {
-            const {variationValues} = variant
-            // update the url with the new product id and variation values when the variant changes
-            const updatedUrl = rebuildPathWithParams(`${location.pathname}${location.search}`, {
-                ...variationValues,
-                pid: variant.productId
-            })
-            history.replace(updatedUrl)
-        }
-    }, [variant])
```

Comment:
This is actually no longer necessary, since there's already a similar, newer logic over at ProductView component. When you click on a Swatch, the page url will be updated with the `href` value here: https://github.com/SalesforceCommerceCloud/pwa-kit/blob/22a6aed44f304e2bd048f1e675eb594d2905cc82/packages/template-retail-react-app/app/components/product-view/index.jsx#L447

Furthermore, `pid` url param is not needed on the Cart page anyways.